File: 二叉树/中根遍历.java
public int[] inorderTraversal(root) {
  BiTreeNode T = root;
  int[] result = new int[];
  if (T != null) {
    LinkStack S = new LinkStack();
    S.push(T);
    while (!S.isEmpty()) {
      while (S.peek() != null) {
        S.push(((BiTreeNode)S.peek()).lchild);
      };
      S.pop();
      if (!S.isEmpty()) {
        T = (BiTreeNode)S.pop();
        result.push(T.val);
        S.push(T.rchild);
      }
    }
  }

  return result;
}

File: 二叉树/中根遍历 .js
// 每个结点都会被访问一次
const inorderTraversal = root => {
  const result = []
  const stack = []
  if (root) {
    let T = root
    while (stack.length) {
      while (stack[stack.length - 1]) {
        stack.push(stack[stack.length - 1].lchild)
      }
      stack.pop()
      if (stack.length) {
        T = stack.pop()
        result.push(T.val)
        stack.push(T.rchild)
      }
    }
  }

  return result
}

